Added GitHub Actions CI

tdulcet opened the pull request:
~**Edit**: This PR is now very outdated, but I would be happy to update it if there is ever interest in adding CI. Travis CI is no longer free, so we would need to use GitHub Actions CI instead.~

- Added GitHub Actions CI (continuous integration).
  - Tests on Ubuntu 22.04.
  - Runs the ShellCheck linter. Bugs fixed in #1457.
  - Run the Ruff Python linter. Some errors fixed in #2343.
    - Config file added in #2473.
  - Ready to run the #1039 test suite, once it is merged. Just uncomment the three lines.

This pull request was originally part of #1435.

JoshData commented:
Great.

Can you explain what's going on in some of the steps? It would be good to add some comments to the .travis.yml file. Also why the change to certbot?

tdulcet committed:
Added GitHub Actions CI.

tdulcet commented:
@JoshData - Now that Travis CI is no longer free, I updated this PR to use GitHub Actions CI and test on Ubuntu 22.04. I also added comments to the workflow file as requested.
